modules/rest_api/controllers/services/rest.php

```diff
@@ -1533,6 +1594,278 @@ private function esToCsv($itemList, $handle) {
     };
   }
 
+  /**
+   * Special field handler returns an empty string. This is useful
+   * where the output CSV must contain a column to which no
+   * useful data can be mapped.
+   *
+   * @param array $doc
+   *   Elasticsearch document.
+   *
+   * @return string
+   *   Empty string.
+   */
+  private function esGetSpecialFieldBlank(array $doc) {
+    return '';
+  }
+
+  /**
+   * Special field handler to format record ID.
+   *
+   * @param array $doc
+   *   Elasticsearch document.
+   *
+   * @return string
+   *   Formatted ID value.
+   */
+  private function esGetSpecialFieldId(array $doc, array $params) {
```

Comment:
This feels very specific to warehouse1's setup so we should try not to include in core code. Perhaps an option is to have a prefix format specifier, so the prefix can be added to $doc['id'] (which is just the numeric ID)?

Reply:
If we added 'BRC1' to index and another warehouse got added to the index it would be confusing - though it would still serve because this is only for backward compatibility with previous downloads which had only ever used BRC1. Thinking about it now, I can't see why this is really needed because records from new and old downloads can just be matched on ID. Shall I go back to @kitenetter to see if this is really needed at all?

Reply:
I think the intention was that the record would have a "guid" that identifies it if it gets out into the wild, e.g. on the NBN atlas. A number on its own won't do that. I see your point though - if we are trying to do this properly, it's some sort of mapping that is required, not just a static prefix. Shame I didn't spot this problem coming when the warehouse was setup - but I wonder if a better solution is to replace brc1| with iBRC across the entire Elasticsearch index, as there are bulk APIs that can do this sort of thing. The Logstash config files would also need a small change, as would the prefix stored in config on the iRecord verification page. If we then have a 2nd warehouse feed into ES, as long as it has it's own unique prefix it should be OK.
Let me know what you think. Maybe worth checking with Martin if anything depends on the brc1| prefix on the ES instance currently.

Reply:
As far as I'm aware there are no critical dependencies for the "brc1|" prefix, because it has not been included in any standard downloads or exports (although it has presumably appeared in the custom ES work for National Trust and RSPB?). Standardising on "iBRC" for all records to date seems to me the best option as it maintains compatibility with most existing data downloads.
@johnvanbreda do you think we should check with, or at least warn, NT and RSPB that this is changing?
In fact we only export the ID number, without the iBRC prefix, to NBN Atlas, so the original intention to have a guid has been lost. Which could be an argument for dropping the prefix altogether ... but probably best to retain it for the backwards-compatible download.

Reply:
Thats fine with me. I'll remove this special field then.

Reply:
Special field removed.

Reply:
@kitenetter before I change anything in the index, will you ask NT/RSPB or shall I?

Reply:
@johnvanbreda I have asked, will report back.

Reply:
@johnvanbreda NT and RSPB are both using the "brc1|" RecordKey at the moment, but both are happy to change to "iBRC". They request that we give them a day or two's notice of when the change will happen.

Reply:
@kitenetter this turned out to be a bit more complex than I thought, since the brc1|<n> key value is the Elasticsearch document ID so it can't be simply updated. I have documented the steps in https://github.com/BiologicalRecordsCentre/iRecord/issues/938 for safekeeping and there is a Logstash script which actually drops the old document and replaces it with a new one with the correct ID.

So, shall I start this in 2 days (Friday) so that you can give NT & RSPB enough notice?

Reply:
@burkmarr unless I'm not reading this pull request correctly doesn't it still contain the special field code for esGetSpecialFieldId?

Reply:
Yes sorry @johnvanbreda - I'd removed the call to it, but not the function. I've done it now.

Reply:
@johnvanbreda I've just emailed NT and RSPB to let them know the change will be happening, so it's ready to go when you are.